src/finfield.h

```diff
@@ -148,17 +152,38 @@ extern  Obj             TypeFF0;
 **  Values of  elements  of  small  finite  fields  are  represented  by  the
 **  logarithm of the element with respect to the root plus one.
 **
-**  Since small finite fields contain at most 65536 elements,  the value fits
-**  into a 'UInt2'.
+**  Depending on the configuration, this type may be a UInt2 or UInt4.
+**  This type is actually defined in gen/ffdata.h  by etc/ffgen.c
+*/
+
+/****************************************************************************
 **
-**  It may be possible to change this to 'UInt4' to allow small finite fields
-**  with more than than  65536 elements.  The macros  and have been coded  in
-**  such a  way that they work  without problems.  The exception is 'POW_FFV'
-**  which  will only work if  the product of integers  of type 'FFV' does not
-**  cause an overflow.  And of course the successor table stored for a finite
-**  field will become quite large for fields with more than 65536 elements.
+*F  PROD_FFV(<a>,<b>,<f>) . . . . . . . . . . . product of finite field value
+**
+**  'PROD_FFV' returns the product of the two finite field values <a> and <b>
+**  from the finite field pointed to by the pointer <f>.
+**
+**  Note that 'PROD_FFV' may only be used if the  operands are represented in
+**  the  same finite field.  If you  want to multiply  two elements where one
+**  lies in a subfield of the other use 'ProdFFEFFE'.
+**
+**  If one of the values is 0 the product is 0.
+**  If $a+b <= o$ we have $a * b ~ z^{a-1} * z^{b-1} = z^{(a+b-1)-1} ~ a+b-1$
+**  otherwise   we   have $a * b ~ z^{(a+b-2)-(o-1)} = z^{(a+b-o)-1} ~ a+b-o$
 */
-typedef UInt2           FFV;
+
+static inline FFV PROD_FFV(FFV a, FFV b, const FFV * f)
```

Comment:
Is there a reason this (and its comment) were moved from before QUO_FFV to before SUM_FFV? This makes the diff bigger and harder to see what changed.